sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6046,8 +6047,7 @@ pi_result _pi_queue::synchronize() {
             (ImmCmdList->first, zeEvent, 0, nullptr));
     ZE_CALL(zeHostSynchronize, (zeEvent));
     Event->Completed = true;
-    Event->ZeCommandList = nullptr;
-    PI_CALL(Event->cleanup(Queue));
```

Comment:
when will this event be cleaned up then?

Reply:
This event doesn't need a cleanup. It is created using EventCreate and not using CreateEventAndAssociateQueue, i.e. it doesn't have associated queue, kernel or dep events. It is created and released inside of this scope and doesn't go outside, so no cleanup is needed and it can be just released.